stumpy/core.py

```diff
@@ -2494,3 +2494,40 @@ def _select_P_ABBA_value(P_ABBA, k, custom_func=None):
             MPdist = partition[k]
 
     return MPdist
+
+
+@njit(parallel=True)
+def _merge_topk_PI(PA, PB, IA, IB):
+    """
+    Merge two top-k matrix profiles PA and PB, and update PA (in place) while
+    prioritizing values of PA in ties. Also, update IA accordingly.
+
+    Parameters
+    ----------
+    PA : numpy.ndarray
+        a (top-k) matrix profile
+
+    PB : numpy.ndarray
+        a (top-k) matrix profile
```

Comment:
Need to add that this array must be sorted in ascending order and needs to be the same shape as `PA`!